pkg/azure/ocpgwdeployer.go

```diff
@@ -125,7 +125,12 @@ func (d *ocpGatewayDeployer) Deploy(input api.GatewayDeployInput, status reporte
 	}
 
 	if d.dedicatedGWNode {
-		err = d.deployDedicatedGWNode(machineSets, gatewayNodesToDeploy, status)
+		image, imageErr := d.msDeployer.GetWorkerNodeImage(nil, nil, d.InfraID)
+		if imageErr != nil {
+			return errors.Wrap(err, "error retrieving worker node image")
```

Comment:
```suggestion
		image, err := d.msDeployer.GetWorkerNodeImage(nil, nil, d.InfraID)
		if err != nil {
			return errors.Wrap(err, "error retrieving worker node image")
```

Reply:
This lead to linting error 

pkg/azure/ocpgwdeployer.go:133:3: ineffectual assignment to err (ineffassign)
		err = d.deployDedicatedGWNode(machineSets, gatewayNodesToDeploy, image, status)

Hence reverting, Will fix the issue in  line 132.